python/instrumentation/openinference-instrumentation-anthropic/tests/openinference/anthropic/test_instrumentor.py

```diff
@@ -0,0 +1,441 @@
+    assert attributes.get(METADATA, None)
+    assert attributes.get(TAG_TAGS, None)
+    assert attributes.get(LLM_PROMPT_TEMPLATE, None)
+    assert attributes.get(LLM_PROMPT_TEMPLATE_VERSION, None)
+    assert attributes.get(LLM_PROMPT_TEMPLATE_VARIABLES, None)
+
+
+def test_anthropic_instrumentation_completions(
+    tracer_provider: TracerProvider,
+    in_memory_span_exporter: InMemorySpanExporter,
+    setup_anthropic_instrumentation: Any,
+    session_id: str,
+    user_id: str,
+    metadata: Dict[str, Any],
+    tags: List[str],
+    prompt_template: str,
+    prompt_template_version: str,
+    prompt_template_variables: Dict[str, Any],
+) -> None:
+    client = Anthropic(api_key="fake")
+    client.completions._post = _mock_post_generation  # type: ignore[assignment]
+
+    with using_attributes(
+        session_id=session_id,
+        user_id=user_id,
+        metadata=metadata,
+        tags=tags,
+        prompt_template=prompt_template,
+        prompt_template_version=prompt_template_version,
+        prompt_template_variables=prompt_template_variables,
+    ):
+        client.completions.create(
+            model="claude-2.1",
+            prompt="How does a court case get to the Supreme Court?",
+            max_tokens_to_sample=1000,
+        )
+
+    spans = in_memory_span_exporter.get_finished_spans()
+
+    assert spans[0].name == "Completions"
+    attributes = dict(spans[0].attributes or {})
+
+    assert attributes.get(OPENINFERENCE_SPAN_KIND) == "LLM"
+    assert attributes.get(f"{LLM_OUTPUT_MESSAGES}.0.{MESSAGE_CONTENT}") == "idk"
+    assert attributes.get(f"{LLM_OUTPUT_MESSAGES}.0.{MESSAGE_ROLE}") == "assistant"
+
+
+def test_anthropic_instrumentation_messages(
+    tracer_provider: TracerProvider,
+    in_memory_span_exporter: InMemorySpanExporter,
+    setup_anthropic_instrumentation: Any,
+    session_id: str,
+    user_id: str,
+    metadata: Dict[str, Any],
+    tags: List[str],
+    prompt_template: str,
+    prompt_template_version: str,
+    prompt_template_variables: Dict[str, Any],
+) -> None:
+    client = Anthropic(api_key="fake")
+    client.messages._post = _mock_post_messages  # type: ignore[assignment]
+
+    client.messages.create(
+        max_tokens=1024,
+        messages=[
+            {
+                "role": "user",
+                "content": "Hello!",
+            }
+        ],
+        model="claude-3-opus-20240229",
+    )
+
+    spans = in_memory_span_exporter.get_finished_spans()
+
+    assert spans[0].name == "Messages"
+    attributes = dict(spans[0].attributes or {})
+
+    assert attributes.get(OPENINFERENCE_SPAN_KIND) == "LLM"
+    assert attributes.get(f"{LLM_OUTPUT_MESSAGES}.0.{MESSAGE_CONTENT}") == "idk"
+    assert attributes.get(f"{LLM_OUTPUT_MESSAGES}.0.{MESSAGE_ROLE}") == "assistant"
+    assert isinstance(attributes.get(LLM_TOKEN_COUNT_PROMPT), int)
+    assert isinstance(attributes.get(LLM_TOKEN_COUNT_COMPLETION), int)
+
+
+@pytest.mark.asyncio
```

Comment:
Wasn't expecting to see the decorator here. Does the test run without?

Reply:
no, pytest ignores it